src/TimeModeling/Types/GeometryStructure.jl

```diff
@@ -161,23 +161,22 @@ end
 function Geometry(data::SegyIO.SeisBlock; key="source", segy_depth_key="")
     src = get_header(data,"FieldRecord")
     nsrc = length(unique(src))
+    gt = Vector{Float32}
     if key=="source"
         isempty(segy_depth_key) && (segy_depth_key="SourceSurfaceElevation")
         params = ["SourceX","SourceY",segy_depth_key]
-        gt = Float32
```

Comment:
I still don't get why you need that and all the other changes seem to follow from it.